Docs/polytopes (#611)
oscar-system/Oscar.jl: docs/src/PolyhedralGeometry/Polyhedra/pg_polyhedra_polymake.md

```@meta
CurrentModule = Oscar
```

```@setup oscar
using Oscar
```

```@contents
Pages = ["pg_polyhedra_polymake.md"]
```

# `Polyhedron` and `polymake`'s `Polytope`

> Many polyhedral computations are done through `polymake`.

Reviewer: The user may not know what "polymake" is at this point. I'd suggest to at least insert one brief sentence explaining it, and perhaps a reference to (to be added) entry in the bibliography for polymake, and/or a link to the polymake homepage?

This is visible in the structure `Polyhedron` via a pointer `pm_polytope` to the corresponding `polymake` object.

> This allows to apply all functionality of `polymake` by calling a suitable `Polymake.jl` function on this pointer.

Reviewer: Again, the read may not know anything about `Polymake.jl` here. So perhaps also introduce that?

> To allow both `Oscar`'s and `polymake`'s functionality to be applicable to a polyhedron object, it can be converted back and forth:

Reviewer: I think that's a rather unfortunate way to state things, given that Oscar is built on polymake, via Polymake.jl, while this makes it sound as if they were two separate systems. 

Also, I hope these are not really doing conversions, but rather are about wrapping and unwrapping objects  Polymake.jl.

So, perhaps like this:
```suggestion
Sometimes it can be necessary to directly invoke some `polymake` functions on an Oscar `Polyhedron` object (e.g. because some functionality has not yet been made available via Oscar's interface). In that case, the following two functions allow extracting the underlying `Polymake.jl` object from a `Polyhedron`, respectively wrapping a `Polymalke.jl` object representing a polyhedron into a high-level `Polyhedron` object.
```

```@docs
Polyhedron(::Polymake.BigObject)
pm_polytope(::Polyhedron)
```

> The following shows all the data "known to" a `Polyhedron`.

Reviewer: I don't see how this sentence fits with the example coming next?

Reviewer 2: Please look at the output. https://oscar-system.github.io/Oscar.jl/previews/PR611/PolyhedralGeometry/Polyhedra/pg_polyhedra_polymake/ It displays all properties that polymake already computed of the cube. Probably the wording should be different.

```@repl oscar
C = cube(3)
C.pm_polytope
```

`polymake` allows for an interactive visualization of 3-dimensional polytopes in the browser: `Polymake.visual(C.pm_polytope)`.

> There are several design differences between `polymake` and `Oscar`.

Reviewer: ... And... ? Are you going to tell us? Or is the "warning" coming next part of this message? If so, it should be made clear, e.g.:
```suggestion
There are several design differences between `polymake` and `Oscar`. The most important are the following:
```
But actually, it might be better to either ditch this sentence, or move it into that "warning"...

!!! warning
    Polyhedra in `polymake` and `Polymake.jl` use homogeneous coordinates. The polyhedra in `Oscar` use affine coordinates.
	Indices in `polymake` are zero-based, whereas in `Oscar` they are one-based.

The next example shows a purely combinatorial construction of a polytope (here: a square).

> Inspite of being given no coordinates, we can see that this is a simple polytope; i.e., each vertex is contained in dimension many facets.

Reviewer: ```suggestion
In spite of being given no coordinates, `polymake` can check for us that this is a simple polytope; i.e., each vertex is contained in dimension many facets.
```

```@repl oscar
Q = Polymake.polytope.Polytope(VERTICES_IN_FACETS=[[0,2],[1,3],[0,1],[2,3]]);
Q.SIMPLE
```

> However, without coordinates, e.g., computing the volume cannot work:

Reviewer: ```suggestion
However, without coordinates, some operations such as computing the volume cannot work:
```
